lib/puppet/functions/deprecation.rb

```diff
@@ -8,6 +8,14 @@
   end
 
   def deprecation(key, message)
+    if defined? Puppet::Pops::PuppetStack.stacktrace()
+      stacktrace = Puppet::Pops::PuppetStack.stacktrace()
+      file = stacktrace[0]
+      line = stacktrace[1]
+      output_message = "#{message} at #{file}:#{line}"
+    else
+      output_message = message
```

Comment:
true. as I said on the original PR, re-using the `message` var would obviate the need to modify the code below, and - in this case - also remove the need fo the `else`.